src/preamble.js

```diff
@@ -144,8 +144,14 @@ var toC = {
   'string': JSfuncs['stringToC'], 'array': JSfuncs['arrayToC']
 };
 
+function returnToType(returnType, ret) {
```

Comment:
how about `convertReturnValue` for the name?

could this be nested in `ccall`? otherwise it seems generally useful, which I don't think it is.